src/Angor/Avalonia/AngorApp/Composition/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,94 @@
+using Angor.UI.Model;
+using Angor.UI.Model.Implementation;
+using Angor.UI.Model.Implementation.Projects;
+using AngorApp.Core;
+using AngorApp.Design;
+using AngorApp.Sections;
+using AngorApp.Sections.Browse;
+using AngorApp.Sections.Founder;
+using AngorApp.Sections.Home;
+using AngorApp.Sections.Portfolio;
+using AngorApp.Sections.Shell;
+using AngorApp.Sections.Wallet;
+using AngorApp.UI.Services;
+using Avalonia.Controls.Notifications;
+using Microsoft.Extensions.DependencyInjection;
+using Zafiro.Avalonia.Controls.Navigation;
+using Zafiro.Avalonia.Dialogs;
+using Zafiro.Avalonia.Services;
+using Zafiro.UI;
+
+namespace AngorApp.Composition;
+
+public static class ServiceCollectionExtensions
+{
+    public static IServiceCollection AddUIServices(this IServiceCollection services, Control parent)
+    {
+        var topLevel = TopLevel.GetTopLevel(parent);
+
+        return services
+            .AddSingleton<ILauncherService>(_ => new LauncherService(topLevel!.Launcher))
+            .AddSingleton<IDialog, DesktopDialog>()
+            .AddSingleton<IActiveWallet, ActiveWallet>()
+            .AddSingleton<INotificationService>(_ => new NotificationService(
+                new WindowNotificationManager(topLevel)
+                {
+                    Position = NotificationPosition.BottomRight
+                }
+            ))
+            .AddSingleton<UIServices>(sp => new UIServices(
+                sp.GetRequiredService<ILauncherService>(),
+                sp.GetRequiredService<IDialog>(),
+                sp.GetRequiredService<INotificationService>(),
+                sp.GetRequiredService<IActiveWallet>()
+            ));
+    }
+
+    public static IServiceCollection AddUIModelServices(this IServiceCollection services)
+    {
+        return services
+            .AddSingleton<IWalletProvider, WalletProviderDesign>()
+            .AddSingleton<IWalletBuilder, WalletBuilderDesign>()
+            .AddSingleton<IWalletFactory, WalletFactory>()
+            .AddSingleton<IProjectService>(sp =>
+            {
+                var loggerFactory = LoggerConfig.CreateFactory();
+                return new ProjectService(
+                    DependencyFactory.GetIndexerService(loggerFactory),
```

Comment:
normally we do not pass the logger factory to services do we?

Reply:
Same as https://github.com/block-core/angor/pull/258/files/fdc6a99058d2550d8813aa81ee87779e3b2022df#r1953160586. Will fix!